docling/backend/docling_parse_v4_backend.py

```
@@ -142,6 +175,28 @@ def __init__(self, in_doc: "InputDocument", path_or_stream: Union[BytesIO, Path]
+            target=_page_worker_main,
+            args=(self._req_queue, self._resp_queue, worker_target),
+            daemon=True,
+        )
+        self._worker.start()
+        try:
+            status, msg = self._resp_queue.get(timeout=30)
+            if status != "ready":
+                raise RuntimeError(f"docling-parse worker init failed: {msg}")
+        except queue.Empty:
+            self._worker.kill()
+            raise RuntimeError("docling-parse worker timed out during init.")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)